### ISSUE TITLE
Full ERC-681 support in scanned QR code

### ISSUE THREAD

**Issue by CodeSandwich** [walleth/walleth#475]
When scanning QR codes with payment details, the app doesn't follow ERC-681 in some details:

- [ ] `ETH` unit is not supported, e.g. `value=1+ETH`
- [ ] custom units are not supported, e.g. `value=1+MyUnit`
- [ ] `gasLimit` is not supported and treated as an invalid data type name, e.g. `gasLimit=100000`
- [ ] `gasPrice` is not supported and treated as an invalid data type name, e.g. `gasPrice=1e12`
- [ ] `gas` is treated as `gasPrice` instead of `gasLimit`, which is against the spec

Tested with Walleth 0.48.2

Reference: [EIP-681](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-681.md)

**Comment by ligi** [comment]
Thanks for the great report! I currently attack the last 3 issues. With the values I am actually not sure - this was added as an afterthought to 681 and I actually see no real use case for this. What are values for MyUnit instead of ETH? WEI?

**Comment by CodeSandwich** [comment]
Yeah, both ETH and MyUnit look absolutely useless, I've included them here only for completeness sake.
